tests/foreman/ui/test_ansible.py

```diff
@@ -785,3 +794,125 @@ def test_positive_schedule_recurring_hostgroup_job(self):
+            5. Click the Schedule Job button.
+            6. Verify that the Ansible roles has been Scheduled.
+
+        :expectedresults: The role is successfully scheduled with provisioned host.
+
+        :BZ: 2025523
+
+        :customerscenario: true
+        """
+        SELECTED_ROLE = 'theforeman.foreman_scap_client'
+        host_name = gen_string('alpha').lower()
+        guest_os_names = 'Red Hat Enterprise Linux 8 (64 bit)'
+        data_store_summary = [
+            h
+            for h in vmwareclient.get_obj_list(vim.Datastore)
+            if h.host and h.name == settings.vmware.datastore
+        ][0].summary
+        uncommitted = data_store_summary.uncommitted or 0
+        capacity = _get_normalized_size(data_store_summary.capacity)
+        free_space = _get_normalized_size(data_store_summary.freeSpace)
+        prov = _get_normalized_size(
+            data_store_summary.capacity + uncommitted - data_store_summary.freeSpace
+        )
+        storage_data = {
+            'storage': {
+                'disks': [
+                    {
+                        'data_store': f'{settings.vmware.datastore} (free: {free_space}, prov: {prov}, total: {capacity})'
+                    }
+                ]
+            }
+        }
+        network_data = {
+            'network_interfaces': {
+                'nic_type': VMWARE_CONSTANTS['network_interface_name'],
+                'network': f'VLAN {settings.provisioning.vlan_id}',
+            }
+        }
+        target_sat.cli.Settings.set({'name': "remote_execution_connect_by_ip", 'value': "yes"})
+        with target_sat.ui_session() as session:
+            session.ansibleroles.import_all_roles()
+            session.location.select(module_location.name)
+            session.organization.select(module_sca_manifest_org.name)
+            assert (
+                session.ansibleroles.import_all_roles() == session.ansibleroles.imported_roles_count
+            )
+            session.hostgroup.assign_role_to_hostgroup(
+                module_vmware_hostgroup.name, {'ansible_roles.resources': SELECTED_ROLE}
+            )
+            assert SELECTED_ROLE in session.hostgroup.read_role(
+                module_vmware_hostgroup.name, SELECTED_ROLE
+            )
+            session.computeresource.update_computeprofile(
+                entity_name=module_vmware_cr.name,
+                compute_profile=COMPUTE_PROFILE_SMALL,
+                values={
+                    'provider_content.memory': '6000',
+                    'provider_content.cluster': settings.vmware.cluster,
+                    'provider_content.guest_os': guest_os_names,
+                    'provider_content.storage': [value for value in storage_data.values()],
+                    'provider_content.network_interfaces': [
+                        value for value in network_data.values()
+                    ],
+                },
+            )
+            session.host.create(
```

Comment:
host creation can be done via API.

Reply:
this is UI provisioning test, so it should be done with UI

Reply:
The host creation does not need to be done in UI even if it is UI test. Anyway, it was non-blocker comment.